direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaWriter.java

```diff
@@ -61,12 +59,14 @@ public void write(StreamElement data, CommitCallback callback) {
       producer.send(
           toWrite,
           (metadata, exception) -> {
-            log.debug(
-                "Written {} to topic {} offset {} and partition {}",
-                data,
-                metadata.topic(),
-                metadata.offset(),
-                metadata.partition());
+            if (metadata != null) {
+              log.debug(
```

Comment:
Hm, this should probably log even if metadata is null. Otherwise it might be confusing. We might replace the values with sensible defaults (-1, etc). Topic can be obtained from URI.

Reply:
Null is returned just in some specific situation when kafka returns error. In that case this log message doesn't make any sense so IMHO it's ok to have this if.

Reply:
OK, makes sense.